tests/unit_tests/ode_solver_unit_test/explicit_ode_solver.cpp

```diff
@@ -0,0 +1,220 @@
+                dealii::Triangulation<dim>::smoothing_on_coarsening));
+
+    int testfail = 0;
+
+    const double left = 0.0;
+    const double right = 2.0;
+    const bool colorize = true;
+    const int n_refinements = 5;
+
+    //generating grid
+    dealii::GridGenerator::hyper_cube(*grid, left, right, colorize);
+    //setting periodic BC 
+    std::vector<dealii::GridTools::PeriodicFacePair<typename Triangulation::cell_iterator> > matched_pairs;
+    if (dim == 2){
+        dealii::GridTools::collect_periodic_faces(*grid,0,1,0,matched_pairs);
+        dealii::GridTools::collect_periodic_faces(*grid,2,3,1,matched_pairs);
+    }
+    else if (dim == 1){
+        dealii::GridTools::collect_periodic_faces(*grid,
+                0, //left
+                1, //right
+                0, //periodic in x-direction
+                matched_pairs);
+    }
+
+    grid->add_periodicity(matched_pairs);
+    grid->refine_global(n_refinements);
+
+    //default parameters
+    dealii::ParameterHandler parameter_handler;
+    PHiLiP::Parameters::AllParameters::declare_parameters (parameter_handler); // default fills options
+    PHiLiP::Parameters::AllParameters all_parameters;
+    all_parameters.parse_parameters (parameter_handler); // copies stuff from parameter_handler into all_parameters
+
+    //ode solver parameters
+    all_parameters.ode_solver_param.ode_solver_type = PHiLiP::Parameters::ODESolverParam::ODESolverEnum::explicit_solver;
+    all_parameters.ode_solver_param.nonlinear_max_iterations = 500;
+    all_parameters.ode_solver_param.print_iteration_modulo = 100;
+    
+    const int rk_order = 3;
+    double expected_order = -double(rk_order);
+    double order_tolerance = 0.1; 
+    all_parameters.ode_solver_param.runge_kutta_order = rk_order;
+
+    const double adv_speed_x = 1.0, adv_speed_y = 0.0;
+    all_parameters.manufactured_convergence_study_param.manufactured_solution_param.advection_vector[0] = adv_speed_x; //x-velocity
+    all_parameters.manufactured_convergence_study_param.manufactured_solution_param.advection_vector[1] = adv_speed_y; //y-velocity
+
+    int n_time_calculations = 4;
+    const double dt_init = 2.5E-3;
+    double dt = dt_init;
+    const double refine_ratio = 0.5;
+    const double finalTime = 1.0;
+
+    dealii::ConvergenceTable convergence_table;
+
+    double L2_error_old = 0;
+    dealii::Vector<double> L2_error_conv_rate(n_time_calculations);
+
+    for (int refinement = 0; refinement < n_time_calculations; ++refinement){
+
+        //initial_time_step is not modified by explicit ODE solver
+        all_parameters.ode_solver_param.initial_time_step = dt;
+        std::cout << "Using time step = " << dt << std::endl;
+        std::cout << "refinement = " << refinement << std::endl;
+
+        all_parameters.ode_solver_param.output_solution_every_x_steps = int(finalTime/dt/10.0); //output 10 vtk files (if dt reaches finalTime exactly)
```

Comment:
there's also a parameter, `output_solution_every_dt_time_intervals`, that you could take advantage of which would be more intuitive, replacing this line by: `all_parameters.ode_solver_param.output_solution_every_dt_time_intervals = int(finalTime/10.0)`

Reply:
Thanks for the suggestion, I will change this.